contrib/ci/dependencies.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/python3
+#
+# Copyright (C) 2017-2018 Dell, Inc.
+# Copyright (C) 2020 Intel, Inc.
+#
+# SPDX-License-Identifier: LGPL-2.1+
+#
+import os
+import sys
+import argparse
+import xml.etree.ElementTree as etree
+
+def parse_dependencies(OS, SUBOS, requested_type):
+    deps = []
+    dep = ''
+    directory = os.path.dirname(sys.argv[0])
+    tree = etree.parse(os.path.join(directory, "dependencies.xml"))
+    root = tree.getroot()
+    for child in root:
+        if not "type" in child.attrib or not "id" in child.attrib:
+            continue
+        for distro in child:
+            if not "id" in distro.attrib:
+                continue
+            if distro.attrib["id"] != OS:
+                continue
+            packages = distro.findall("package")
+            for package in packages:
+                if SUBOS:
+                    if not 'variant' in package.attrib:
+                        continue
+                    if package.attrib['variant'] != SUBOS:
+                        continue
+                if package.text:
+                    dep = package.text
+                else:
+                    dep = child.attrib["id"]
+                if child.attrib["type"] == requested_type and dep:
+                    deps.append(dep)
+    return deps
+
+if __name__ == '__main__':
+
+    try:
+        import distro
+        target = distro.linux_distribution()[0]
+    except ModuleNotFoundError:
+        target = None
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-o", "--os",
+                        default=target,
+                        choices=["fedora",
+                                 "centos",
+                                 "flatpak",
+                                 "debian",
+                                 "ubuntu",
+                                 "arch"],
```

Comment:
I'm not crazy about this being hardcoded; but admittedly this changes very rarely and there isn't a place to readily read the supported distros.